src/main/java/org/aszjch/demoapp/domain/article/ArticleServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+package org.aszjch.demoapp.domain.article;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.aszjch.demoapp.domain.article.port.ArticlePublisher;
+import org.aszjch.demoapp.domain.article.port.ArticleRepository;
+import org.aszjch.demoapp.domain.articlefile.ArticleFile;
+import org.aszjch.demoapp.domain.articlefile.port.ArticleFileAssignmentService;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+@RequiredArgsConstructor
+public class ArticleServiceImpl implements ArticleService, ArticleFileAssignmentService {
+
+    private final ArticleRepository repository;
+    private final ArticlePublisher publisher;
+
+    @Override
+    public List<Article> get() {
+        return repository.findAll();
+    }
+
+    @Override
+    public Article create(Article article) {
+        Article saved = repository.save(article);
+        publisher.publish(
+                String.valueOf(new ArticleCreatedDto(saved.getId(), saved.getTitle(), saved.getAuthor(),
+                                                     saved.getAbstractText())));
+        return saved;
+    }
+
+    @Override
+    public Optional<Article> getById(Long id) {
+        return repository.findById(id);
+    }
+
+    @Override
+    public Article update(Long id, Article article) {
+        Article entity = getById(id).orElse(new Article());
```

Comment:
Chcemy robić UPSERT? Czy klasyczek - PUT jako full update tylko?
IMHO możemy zostawić :)

Reply:
Wcześniej rozmawialiśmy o upsert, brzmiał bardzo sensownie, więc czemu nie 😄

Reply:
Niby istnieje możliwość, że ktoś mógłby przywrócić w taki sposób artykuł, który w międzyczasie usunął, ale w kontekście tego projektu raczej nie robi to dużej różnicy 🤔